cdap-app-fabric/src/main/java/co/cask/cdap/gateway/handlers/ProgramLifecycleHttpHandler.java

```diff
@@ -244,6 +244,40 @@ public void getStatus(HttpRequest request, HttpResponder responder,
     responder.sendJson(HttpResponseStatus.OK, status);
   }
 
+  /**
+   * Returns status of a type specified by the type{flows,workflows,mapreduce,spark,services,schedules}.
+   */
+  @GET
+  @Path("/apps/{app-id}/versions/{version-id}/{type}/{id}/status")
+  public void getVersionStatus(HttpRequest request, HttpResponder responder,
```

Comment:
This is to distinguish it from the `getStatus` method above. What method name would you suggest?